Microsoft.Azure.Cosmos/src/Routing/GatewayAddressCache.cs

```diff
@@ -146,11 +156,50 @@ internal class GatewayAddressCache : IAddressCache, IDisposable
                         this.serverPartitionAddressCache.Set(
                             new PartitionKeyRangeIdentity(collection.ResourceId, addressInfo.Item1.PartitionKeyRangeId),
                             addressInfo.Item2);
+
+                        if (openConnectionHandler != null)
+                        {
+                            await this.OpenRntbdChannelsAsync(
+                                addressInfo,
+                                openConnectionHandler);
+                        }
                     }
                 }
             }
         }
 
+        /// <summary>
+        /// Invokes the transport client delegate to open the Rntbd connection
+        /// and establish Rntbd context negotiation to the backend replica nodes.
+        /// </summary>
+        /// <param name="addressInfo">An instance of <see cref="Tuple{T1, T2}"/> containing the partition key id
+        /// and it's corresponding address information.</param>
+        /// <param name="openConnectionHandlerAsync">The transport client callback delegate to be invoked at a
+        /// later point of time.</param>
+        private async Task OpenRntbdChannelsAsync(
+             Tuple<PartitionKeyRangeIdentity, PartitionAddressInformation> addressInfo,
+             Func<Uri, Task> openConnectionHandlerAsync)
+        {
+            foreach (AddressInformation address in addressInfo.Item2.AllAddresses)
+            {
+                DefaultTrace.TraceVerbose("Attempting to open Rntbd connection to backend uri: {0}. '{1}'",
+                    address.PhysicalUri,
+                    System.Diagnostics.Trace.CorrelationManager.ActivityId);
+                try
+                {
+                    await openConnectionHandlerAsync(
+                        new Uri(address.PhysicalUri));
+                }
+                catch (Exception ex)
+                {
+                    DefaultTrace.TraceWarning("Failed to open Rntbd connection to backend uri: {0} with exception: {1}. '{2}'",
+                        address.PhysicalUri,
+                        ex.Message,
```

Comment:
Don't just trace the message, trace the complete exception
```suggestion
                        ex,
```

Reply:
Make sense! Done.